src/main/java/com/ylysenkova/movieland/dao/jdbc/JdbcMovieDao.java

```diff
@@ -120,10 +130,63 @@ public Movie getMovieById(int movieId) {
+
+        logger.info("Inserting movie ={} is started.", movie);
+
+        KeyHolder generatedMovieId = new GeneratedKeyHolder();
+        MapSqlParameterSource sqlParameterSource = new MapSqlParameterSource();
+        sqlParameterSource.addValue("nameRussian", movie.getNameRussian());
+        sqlParameterSource.addValue("nameNative", movie.getNameNative());
+        sqlParameterSource.addValue("yearOfRelease", movie.getYearOfRelease());
+        sqlParameterSource.addValue("description", movie.getDescription());
+        sqlParameterSource.addValue("rating", movie.getRating());
+        sqlParameterSource.addValue("price", movie.getPrice());
+        sqlParameterSource.addValue("picturePath", movie.getPicturePath());
+        namedParameterJdbcTemplate.update(insertMovieSQL, sqlParameterSource, generatedMovieId);
+        for (Genre genre : movie.getGenres()) {
+            sqlParameterSource.addValue("movieId", generatedMovieId.getKey());
+            sqlParameterSource.addValue("genreId", genre.getId());
+            namedParameterJdbcTemplate.update(insertMovieGenreSQL, sqlParameterSource);
+        }
+        for (Country country : movie.getCountries()) {
+            sqlParameterSource.addValue("movieId", generatedMovieId.getKey());
+            sqlParameterSource.addValue("countryId", country.getId());
+            namedParameterJdbcTemplate.update(insertMovieCountrySQL, sqlParameterSource);
+        }
+    }
+
+    @Override
+    public void editMovie(Movie movie) {
+        KeyHolder generatedMovieId = new GeneratedKeyHolder();
+        logger.info("Movie with movie id ={} is updating", movie.getId());
+        MapSqlParameterSource sqlParameterSource = new MapSqlParameterSource();
+        sqlParameterSource.addValue("movieId", movie.getId());
+        sqlParameterSource.addValue("nameRussian", movie.getNameRussian());
+        sqlParameterSource.addValue("nameNative", movie.getNameNative());
+        sqlParameterSource.addValue("yearOfRelease", movie.getYearOfRelease());
+        sqlParameterSource.addValue("description", movie.getDescription());
+        sqlParameterSource.addValue("rating", movie.getRating());
+        sqlParameterSource.addValue("price", movie.getPrice());
+        sqlParameterSource.addValue("picturePath", movie.getPicturePath());
+        namedParameterJdbcTemplate.update(updateMovieSQL, sqlParameterSource);
+        for (Genre genre : movie.getGenres()) {
+            sqlParameterSource.addValue("movieId", generatedMovieId.getKey());
+            sqlParameterSource.addValue("genreId", genre.getId());
+            namedParameterJdbcTemplate.update(insertMovieGenreSQL, sqlParameterSource);
+        }
+        for (Country country : movie.getCountries()) {
+            sqlParameterSource.addValue("movieId", generatedMovieId.getKey());
+            sqlParameterSource.addValue("countryId", country.getId());
+            namedParameterJdbcTemplate.update(insertMovieCountrySQL, sqlParameterSource);
+        }
+    }
```

Comment:
It's bit strange that you're removing countries and genres in respective dao classes but inserting them in MovieDao